internal/worker/authenticationworker/worker.go

```diff
@@ -42,51 +43,65 @@ func NewWorker(st *keyupdater.State, agentConfig agent.Config) (worker.Worker, e
 	if !ok {
 		return nil, errors.NotValidf("machine tag %v", agentConfig.Tag())
 	}
+	keyUpdater := &keyUpdater{
+		st:  st,
+		tag: machineTag,
+	}
+	err := keyUpdater.initializeAuthorizedKeys()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
 	w, err := watcher.NewNotifyWorker(watcher.NotifyConfig{
-		Handler: &keyupdaterWorker{
-			st:  st,
-			tag: machineTag,
-		},
+		Handler: keyUpdater,
 	})
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	return w, nil
+
+	return AuthWorker{
+		NotifyWorker: w,
+	}, nil
 }
 
-// SetUp is defined on the worker.NotifyWatchHandler interface.
-func (kw *keyupdaterWorker) SetUp() (watcher.NotifyWatcher, error) {
+// initializeAuthorizedKeys reads the current ssh keys from the state
+// and the current ssh keys from ~/.ssh/authorized_keys. It removes any
+// ephemeral keys.
+func (kw *keyUpdater) initializeAuthorizedKeys() error {
 	// Record the keys Juju knows about.
 	jujuKeys, err := kw.st.AuthorisedKeys(kw.tag)
```

Comment:
This happens-before the call to `WatchAuthorisedKeys` which will result in missing notifications, but this is OK because the worker is able to catch up after startup.